core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/UpxCompressionBuildStep.java

```diff
@@ -0,0 +1,204 @@
+                .redirectError(ProcessBuilder.Redirect.PIPE);
+        Process process = null;
+        try {
+            process = processBuilder.start();
+            ProcessUtil.streamOutputToSysOut(process);
+            final int exitCode = process.waitFor();
+            if (exitCode != 0) {
+                log.errorf("Command: " + String.join(" ", args) + " failed with exit code " + exitCode);
+                return false;
+            }
+            log.infof("Native executable compressed: %s", executable.getAbsolutePath());
+            return true;
+        } catch (Exception e) {
+            log.errorf("Command: " + String.join(" ", args) + " failed", e);
+            return false;
+        } finally {
+            if (process != null) {
+                process.destroy();
+            }
+        }
+
+    }
+
+    private boolean runUpxInContainer(NativeImageBuildItem nativeImage, NativeConfig nativeConfig) {
+        String level = getCompressionLevel(nativeConfig.compression.level.getAsInt());
+        List<String> extraArgs = nativeConfig.compression.additionalArgs.orElse(Collections.emptyList());
+
+        List<String> commandLine = new ArrayList<>();
+        NativeConfig.ContainerRuntime containerRuntime = nativeConfig.containerRuntime
+                .orElseGet(NativeImageBuildContainerRunner::detectContainerRuntime);
+        commandLine.add(containerRuntime.getExecutableName());
+
+        commandLine.add("run");
+        commandLine.add("--env");
+        commandLine.add("LANG=C");
+        commandLine.add("--rm");
+        commandLine.add("--entrypoint=upx");
+
+        String containerName = "upx-" + RandomStringUtils.random(5, true, false);
+        commandLine.add("--name");
+        commandLine.add(containerName);
+
+        String volumeOutputPath = nativeImage.getPath().toFile().getParentFile().getAbsolutePath();
+        if (SystemUtils.IS_OS_WINDOWS) {
+            volumeOutputPath = FileUtil.translateToVolumePath(volumeOutputPath);
+        } else if (SystemUtils.IS_OS_LINUX) {
+            String uid = getLinuxID("-ur");
+            String gid = getLinuxID("-gr");
+            if (uid != null && gid != null && !uid.isEmpty() && !gid.isEmpty()) {
+                Collections.addAll(commandLine, "--user", uid + ":" + gid);
+                if (containerRuntime == NativeConfig.ContainerRuntime.PODMAN) {
+                    // Needed to avoid AccessDeniedExceptions
+                    commandLine.add("--userns=keep-id");
+                }
+            }
+        }
+
+        Collections.addAll(commandLine, "-v",
+                volumeOutputPath + ":" + NativeImageBuildStep.CONTAINER_BUILD_VOLUME_PATH + ":z");
+
+        commandLine.add(nativeConfig.builderImage);
+        commandLine.add(level);
+        commandLine.addAll(extraArgs);
+
+        commandLine.add(nativeImage.getPath().toFile().getName());
+
+        log.infof("Compression native executable using: %s", String.join(" ", commandLine));
```

Comment:
```suggestion
        log.infof("Compress native executable using: %s", String.join(" ", commandLine));
```

Reply:
Actually I think just printing the joined command line list would be enough here